src/delivery_group.rs

```diff
@@ -107,11 +101,19 @@ fn candidates(
         .sorted_by(|lhs, rhs| lhs.prefix.cmp_distance(&rhs.prefix, target_name))
         .map(|info| (&info.prefix, info.elders.len(), info.elders.values()));
 
-    let mut dg_size = 0;
+    let mut dg_size = ELDER_MAJORITY;
     let mut nodes_to_send = Vec::new();
     for (idx, (prefix, len, connected)) in sections.enumerate() {
         nodes_to_send.extend(connected.cloned());
-        dg_size = delivery_group_size(len);
+        // If we don't have enough contacts send to as many as possible
+        // up to majority of Elders
+        dg_size = cmp::min(len, dg_size);
+        if len < ELDER_MAJORITY {
+            warn!(
+                "Delivery group only {:?} when it should be {:?}",
```

Comment:
`when it should be` sounds like this case is a `definitely wrong situation`, and shall terminate and return
However, such case may happen and we do continue.
Maybe change to : `Delivery group only {:?} when expected to be {:?}` ?

Reply:
It's an area we should talk more about. The consts assume we have that amount of Elders and a fixed majority. So when there are less and the majority is still >50% of remaining nodes then is it still majority? Tricky edge case. I assume we keep going, especially now as we have at least once delivery and lazy messaging. 
I will make this change though @maqi  as what you suggest reads better for sure.